filters/log/log.go

```diff
@@ -208,16 +221,21 @@ func (ual *unverifiedAuditLog) Request(ctx filters.FilterContext) {
 			return
 		}
 
-		var j jwtMidPart
+		var j map[string]interface{}
 		err = json.Unmarshal(sDec, &j)
 		if err != nil {
 			return
 		}
-		req.Header.Add(UnverifiedAuditHeader, cleanSub(j.Sub))
+
+		if k, ok := j[ual.TokenKey]; ok {
+			if v, ok2 := k.(string); ok2 {
+				req.Header.Add(UnverifiedAuditHeader, cleanSub(v))
```

Comment:
Does cleanSub work for non-sub parts?

Reply:
It's some kind of sanitizing regex that the value needs to match:
`re = regexp.MustCompile("^[a-zA-z0-9_/:?=&%@.#-]*$")`

Looks like it should be good for anything that we'd want though...

Reply:
Ah I see it just has to match `^[a-zA-z0-9_/:?=&%@.#-]*$`, but then it defaults the header to <invalid-sub> when it may not be sub they were looking for

Reply:
Maybe this should be added to the documentation, that the data has to match this regular expression.

Reply:
Adding key specific defaults might be a bit overkill. If the key is not present, it will not use the default. So it would only end up being set if the value of the dynamic was invalid as per the regex.... 🤔

Reply:
@szuecs this particular filter isnt documented. Not sure what the reason is for that.
I'm happy to throw in some documentation as part of this PR though

Reply:
Yes, sorry for that. It was my fault not to document it in the first place.

Reply:
We should have defaults to not break compatibility. I don't know if someone is using it, but we want to do our best to not break filter configurations out there.

Reply:
Ok @szuecs / @herojan 
Documentation added for the `unverifiedAuditLog` filter.
I was careful to try and not break backward compatibility. If there is no parameter passed to the filter, as per previously, it will default to using the sub key in the lookup...